mapie/conformity_scores/sets/naive.py

```diff
@@ -0,0 +1,506 @@
+            Prediction set for each observation and each alpha.
+
+        y_pred_index_last: NDArray of shape (n_samples, threshold)
+            Index of the last included label.
+
+        y_pred_proba_cumsum: NDArray of shape (n_samples, n_classes)
+            Cumsumed probability of the model in the original order.
+
+        y_pred_proba_last: NDArray of shape (n_samples, 1, threshold)
+            Last included probability.
+
+        threshold: NDArray of shape (n_alpha,) or shape (n_samples_train,)
+            Threshold to compare with y_proba_last_cumsum, can be either:
+
+            - the quantiles associated with alpha values when
+                ``cv`` == "prefit", ``cv`` == "split"
+                or ``agg_scores`` is "mean"
+
+            - the conformity score from training samples otherwise (i.e., when
+            ``cv`` is CV splitter and ``agg_scores`` is "crossval")
+
+        method: str
+            Method that determines how to remove last label in the prediction
+            set.
+
+            - if "cumulated_score" or "aps", compute V parameter
+                from Romano+(2020)
+
+            - else compute V parameter from Angelopoulos+(2020)
+
+        lambda_star: Optional[Union[NDArray, float]] of shape (n_alpha):
+            Optimal value of the regulizer lambda.
+
+        k_star: Optional[NDArray] of shape (n_alpha):
+            Optimal value of the regulizer k.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes, n_alpha)
+            Updated version of prediction_sets with randomly removed labels.
+        """
+        # get cumsumed probabilities up to last retained label
+        y_proba_last_cumsumed = np.squeeze(
+            np.take_along_axis(
+                y_pred_proba_cumsum,
+                y_pred_index_last,
+                axis=1
+            ), axis=1
+        )
+
+        # get the V parameter from Romano+(2020) or Angelopoulos+(2020)
+        vs = self._compute_vs_parameter(
+            y_proba_last_cumsumed,
+            threshold,
+            y_pred_proba_last,
+            prediction_sets
+        )
+
+        # get random numbers for each observation and alpha value
+        random_state = check_random_state(random_state)
+        random_state = cast(np.random.RandomState, random_state)
+        us = random_state.uniform(size=(prediction_sets.shape[0], 1))
+        # remove last label from comparison between uniform number and V
+        vs_less_than_us = np.less_equal(vs - us, EPSILON)
+        np.put_along_axis(
+            prediction_sets,
+            y_pred_index_last,
+            vs_less_than_us[:, np.newaxis, :],
+            axis=1
+        )
+        return prediction_sets
+
+    def get_prediction_sets(
+        self,
+        y_pred_proba: NDArray,
+        conformity_scores: NDArray,
+        alpha_np: NDArray,
+        estimator: EnsembleClassifier,
+        agg_scores: Optional[str] = "mean",
```

Comment:
Can you includ agg_scores and include_last_label to the kwargs as none of them is used for naive conformity score ?

Reply:
Done 👍 (I have removed them because they are useless in the naive method)